src/gengowatcher/browser_detector.py

```
@@ -0,0 +1,360 @@
+
+        except Exception as e:
+            self.logger.debug(f"Error detecting system browser: {e}")
+
+        return None
+
+    def _detect_linux_browser(self) -> Optional[str]:
+        """Detect browser on Linux systems."""
+        try:
+            # Try xdg-settings first (most reliable)
+            result = subprocess.run(
+                ["xdg-settings", "get", "default-web-browser"],
```

Ruff (0.15.0)

[error] Starting a process with a partial executable path

(S607)